simple-dns/src/dns/rdata/txt.rs

```diff
@@ -117,6 +115,38 @@ impl<'a> TryFrom<HashMap<String, Option<String>>> for TXT<'a> {
     }
 }
 
+impl<'a> TryFrom<String> for TXT<'a> {
+    type Error = crate::SimpleDnsError;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        let mut txt = TXT::new();
+
+        let mut start_index = 0;
+        let full_length = value.len();
+
+        while start_index < full_length {
+            let end_index = (start_index + MAX_CHARACTER_STRING_LENGTH).min(full_length);
+
+            let slice = &value[start_index..end_index];
+            txt.add_char_string(slice.to_string().try_into()?);
+
+            start_index = end_index;
+        }
+
+        Ok(txt)
+    }
+}
```

Comment:
This can be simplified if you implement for `&str`, and it would also avoid cloning the data

```suggestion
impl<'a> TryFrom<&'a str> for TXT<'a> {
    type Error = crate::SimpleDnsError;

    fn try_from(value: &'a str) -> Result<Self, Self::Error> {
        let mut txt = TXT::new();

        for v in value.as_bytes().chunks(MAX_CHARACTER_STRING_LENGTH) {
            txt.add_char_string(CharacterString::new(v)?);
        }

        Ok(txt)
    }
}
```